### DISCUSSION TITLE
More specific string info for "Inspect" mode

### DISCUSSION THREAD

**Discussion by benajamin** [silx-kit/h5web#1612] [Ideas]
One of the difficulties new adopters of HDF5 come across is that there are many ways to write strings (fixed vs variable length, null terminated or not). Many tools don't clearly tell what they are reading or writing, and some programs will crash or malfunction if they don't get an expected string type. The hdfview utility is the one that I find most useful for debugging string issues and it would be great if the h5web viewer could also provide this detailed string info in the "inspect" mode and therefore help debug HDF5 string bugs.

**Comment by axelboc** [comment]
Hi @benajamin, thanks for the suggestion.

We already show whether a string type is variable-length or fixed-length by giving the number of characters for fixed-length strings — if this information is not present, then it's a variable-length string. Perhaps we can make this more explicit, though:

![image](https://github.com/silx-kit/h5web/assets/2936402/76b4c284-e118-48d5-bb53-a0b0926840b6)
![image](https://github.com/silx-kit/h5web/assets/2936402/ae110e65-70f3-44e4-8fa6-cb0bafcf6275)

Perhaps we could add `, variable length` or ` (variable length)` in the first case?

The information we currently don't show at all is the [padding](https://api.h5py.org/h5t.html#strings), but we can easily add it. This will require changes in the back-end providers ([h5grove](https://github.com/silx-kit/h5grove/) and [h5wasm](https://github.com/usnistgov/h5wasm) notably). How would you like this information to be displayed (i.e. is there a "wrong" kind of padding; what should the wording be; does it need to appear explicitly on the _Type_ row or can it be hidden away under the _Inspect_ accordion; etc.)

Is any other information missing?

**Reply by benajamin** [reply]
I think that making the string length information more explicit would help users understand that the details you are presenting are deliberate and can be trusted. The string padding info would be great to have but can be hidden behind the _Inspect_ accordion since it is probably only interesting for debugging. I don't think that there is a definite right or wrong type of padding to use, I have just seen incompatibilities between programs reading and writing HDF5 that caused bugs that were difficult to diagnose because the critical padding information was not easy to expose.

I can't think of anything else that is missing. Thanks for making this nice tool!

**Reply by axelboc** [reply]
You should now see "variable length" as well as the string padding in [myHDF5](https://myhdf5.hdfgroup.org/):

![image](https://github.com/silx-kit/h5web/assets/2936402/037fc4bf-249b-42b2-a6ac-ddfe94a12cee)
